src/deluge/model/note/note_row.cpp

```diff
@@ -1959,12 +1959,15 @@ void NoteRow::renderRow(TimelineView* editorScreen, RGB rowColour, RGB rowTailCo
 				}
 			}
 
-			// Or if Note starts exactly on square...
+			// Or if note starts exactly on square...
 			else if (note && note->pos == squareStartPos) {
 				drewNote = true;
-				// this is just colour * (0.25 + 0.75 * velocity) but in fixed point, i.e. it'll go linearly from 0.25
-				// to max brightness with velocity
-				pixel = rowColour.adjustFractional((65 + note->velocity + (note->velocity / 2)) << 8, 255 << 8);
+				float velocity_ratio = (float)(note->velocity) / 127; // velocity range is 1-127
```

Comment:
Please don't use a C-style cast, use either `static_cast<float>(value)` or `std::bit_cast<float>(value)` whether you are casting the float via conversion or preserving bit positions, respectively
```suggestion
				float velocity_ratio = static_cast<float>(note->velocity) / 127.f; // velocity range is 1-127
```